### ISSUE TITLE
Remote and local memory management

### ISSUE THREAD

**Issue by LaRiffle** [OpenMined/PySyft#1478]
### User Story:
1. When running a Fed Learning demo, you're are forced to set `local_worker.is_client_worker = False` otherwise you don't register any objects in `local_worker._objects` and at a point of the demo this raises a key error. Doing this, you store tons of useless tensors that fill the memory without purpose. We should investigate why the training requires to have objects stored in `local_worker._objects`.
2. Remote tensors that are results of intermediate operations are not deleted:
```
x = sy.FloatTensor([1, 2])
x.send(bob) # X1 is stored at Bob, and x points at X1
x = x.abs() # X2 = X1.abs() is stored and x points at X2 => reference to X1 is lost: will never be deleted
```
  This stale memory should be wiped out.

**Comment by iamtrask** [comment]
Hey @bigdata2 - you are almost certainly running into this issue. We're addressing it in the Torch 1.0 Refactor. You can track progress (or help out) by watching this issue https://github.com/OpenMined/PySyft/issues/1701

**Comment by iamtrask** [comment]
garbage collection solved this in torch 1.0
